src/Psr/Tracing/RelayOpenTelemetry.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CacheWerk\Relay\Psr\Tracing;
+
+use Relay\Relay;
+
+use OpenTelemetry\API\Trace\SpanKind;
+use OpenTelemetry\API\Trace\TracerInterface;
+
+class RelayOpenTelemetry
+{
+    /**
+     * The Relay instance.
+     *
+     * @var \Relay\Relay
+     */
+    protected Relay $relay;
+
+    /**
+     * The OpenTelemetry tracer instance.
+     *
+     * @var \OpenTelemetry\API\Trace\TracerInterface
+     */
+    protected TracerInterface $tracer;
+
+    /**
+     * Creates a new `NewRelicRelay` instance.
+     *
+     * @param  \Relay\Relay  $relay
+     * @param  \OpenTelemetry\API\Trace\TracerInterface  $tracer
+     * @return void
+     */
+    public function __construct(Relay $relay, TracerInterface $tracer)
+    {
+        $this->relay = new $relay;
+        $this->tracer = new $tracer;
+    }
+
+    /**
+     * Executes Relay methods inside OpenTelemetry span.
+     *
+     * @param  string  $name
+     * @param  array  $arguments
+     * @return mixed
+     */
+    public function __call(string $name, array $arguments)
+    {
+        $span = $this->tracer->spanBuilder('Relay') // TODO: What's the name?
```

Comment:
This should definely be either `Relay::$name` or just `$name`. Both is fine and is a matter of taste.

I assume that the `$name` has low cardinality / low number of distinct values (up ot few hundreds), because it will be used to group similar operations/spans together. There is some guidance on span names [here](https://uptrace.dev/opentelemetry/span-naming.html#span-names).

Reply:
Great link, thanks!